### PR TITLE
changes dtype default value in read_sae_from_disk()

### LINKED ISSUE #230: [Bug Report] read_sae_from_disk should default to the SAE dtype unless it's overridden
**Describe the bug**
`sae_lens_loader` in `pretrained_sae_loaders.py` currently does not pass in `dtype` into `read_sae_from_disk`, so it always initializes weights using `float32` first.

The dtype is correct at the very end, since `sae` is created/initialized with the correct dtype from cfg_dict. However, ideally we would be using the correct dtype the whole time.

The fix seems to be that `read_sae_from_disk` should 1) make dtype optional and 2) use the dtype specified by the cfg_dict by default, if dtype is not specified. This has the benefit that other methods calling `read_sae_from_disk` would have the correct behavior by default, unless they choose to override.

I'll make a PR for this soon.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/jbloomAus/SAELens/issues) in the repo (**required**)

### PR DESCRIPTION
**Pull request by anthonyduong9**
# Description

Changes `read_sae_from_disk()` so that the `dtype` param is of type `Optional[torch.dtype]` and has a default value of `None`, in which case the function basically sets `dtype` to `cfg_dict["dtype"]`. `dtype` should be `cfg_dict["dtype"]`, unless the caller specifies a different value.

Fixes #230 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have not rewritten tests relating to key interfaces which would affect backward compatibility

### You have tested formatting, typing and unit tests (acceptance tests not currently in use)

- [x] I have run `make check-ci` to check format and linting. (you can run `make format` to format code if needed.)

### Performance Check.

If you have implemented a training change, please indicate precisely how performance changes with respect to the following metrics:
- [ ] L0
- [ ] CE Loss
- [ ] MSE Loss
- [ ] Feature Dashboard Interpretability

Please links to wandb dashboards with a control and test group.

### CODE REVIEW

**Review by chanind — APPROVED** [review]
👍 LGTM
